spec/steps/step/attributes_spec.rb

```diff
@@ -0,0 +1,25 @@
+require "rails_helper"
+
+RSpec.describe Step do
+  describe "#hash_key?" do
+    it "returns true if the attribute is a hash key" do
+      names = [
+        { foo: [] },
```

Comment:
These tests are great! I don't think I understand why we would expect a Hash in one case and a Symbol in another. Could you help explain that? And is there a way we could make those expectations or reasoning clearer in the code?